src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -254,9 +254,32 @@ public override SyntaxNode VisitSimpleLambdaExpression(SimpleLambdaExpressionSyn
             public override SyntaxNode VisitArgument(ArgumentSyntax node)
             {
                 _cancellationToken.ThrowIfCancellationRequested();
+                string inferredName = null;
+                if (node.Parent.IsKind(SyntaxKind.TupleExpression))
+                {
+                    var tuple = (TupleExpressionSyntax)node.Parent;
+                    if (!IsTupleInDeconstruction(tuple))
+                    {
+                        inferredName = node.Expression.TryGetInferredMemberName().ValueText;
+                        if (!CanMakeNameExplicitInTuple(tuple, inferredName))
+                        {
+                            inferredName = null;
+                        }
+                    }
+                }
 
                 var newArgument = (ArgumentSyntax)base.VisitArgument(node);
 
+                if (inferredName != null && node.NameColon == null)
```

Comment:
You only needed the inferred name if nameColon was not null.  So you should move that outer code in here.  (though likely a helper method would be good.  but htat helper method should be called inside this if block.